gtime/forecasting/simple_models.py

```diff
@@ -0,0 +1,354 @@
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, RegressorMixin
+from sklearn.utils.validation import check_is_fitted
+
+
+class NaiveModel(BaseEstimator, RegressorMixin):
+
+    """Naïve model, all predicted values are equal to the most recent available observation.
+
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> import numpy as np
+    >>> from gtime.model_selection import horizon_shift, FeatureSplitter
+    >>> from gtime.forecasting import NaiveModel
+    >>> idx = pd.period_range(start='2011-01-01', end='2012-01-01')
+    >>> np.random.seed(1)
+    >>> df = pd.DataFrame(np.random.random((len(idx), 1)), index=idx, columns=['1'])
+    >>> y = horizon_shift(df, horizon=3)
+    >>> X_train, y_train, X_test, y_test = FeatureSplitter().transform(df, y)
+    >>> m = NaiveModel()
+    >>> m.fit(X_train, y_train).predict(X_test)
+                         y_1       y_2       y_3
+        2011-12-30  0.541559  0.541559  0.541559
+        2011-12-31  0.974740  0.974740  0.974740
+        2012-01-01  0.636604  0.636604  0.636604
+    """
+
+    def fit(self, X: pd.DataFrame, y=None):
+
+        """Fit the estimator.
+
+        Parameters
+        ----------
+        X : pd.DataFrame, shape (n_samples, n_features), train sample, required for compatibility, not used for a naive model.
+
+        y : None
+            Used to store the predict feature names and prediction horizon.
+
+        Returns
+        -------
+        self : NaiveModel
+            Returns self.
+        """
+
+        self._y_columns_ = y.columns
+        self._horizon_ = len(y.columns)
+
+        return self
+
+    def predict(self, X: pd.DataFrame) -> pd.DataFrame:
+
+        """Using the value of each element in ``X`` predicts the rest of the forecast to be equal to it.
+
+        Parameters
+        ----------
+        X: pd.DataFrame, shape (n_samples, 1), required
+            The time series on which to predict.
+
+        Returns
+        -------
+        predictions : pd.DataFrame, shape (n_samples, self._horizon)
+            The output predictions.
+
+        Raises
+        ------
+        NotFittedError
+            Raised if the model is not fitted yet.
+
+        """
+
+        check_is_fitted(self)
+        y_pred = np.broadcast_to(X, (len(X), self._horizon_))
+        predictions = pd.DataFrame(data=y_pred, columns=self._y_columns_, index=X.index)
+
+        return predictions
+
+
+class SeasonalNaiveModel(NaiveModel):
```

Comment:
These methods to not need to inherit from `NaiveModel`, since they are not using any of its methods

Reply:
They use `fit()` for storing basic information about y since it's not observable in `predict`:
```
self._y_columns_ = y.columns
self._horizon_ = len(y.columns)
```
I thought of inheriting some common `predict` commands like 
```
check_is_fitted(self)
predictions = pd.DataFrame(data=np.nan, columns=self._y_columns_, index=X.index)
```
too but in this case it should probably be an abstract parent class for all the models. What do you think?

Reply:
Ah, I see your point.
Yes, I think it is a good idea then to inherit from `NaiveModel`. And as you were saying, it would be even better to have an abstract parent class.

If it does not take too much time to implement that, maybe you can do it

Reply:
Abstract parent class added